pages/api/routine_logs/[routineName].js

```diff
@@ -0,0 +1,17 @@
+import { connectToDatabase, find } from "../../../utils/nedb"
+
+export default async function handler(req, res) {
+    const { routineName } = req.query
+    const { db } = connectToDatabase()
+
+    let exceptions = await find(db.exceptions, {Source: routineName})
+    let warnings = await find(db.warnings, {Source: routineName})
+    let infos = await find(db.infos, {Source: routineName})
+    let plogs = await find(db.plogs, {Source: routineName})
```

Comment:
Maybe `promise.all` might be better here in terms of performance.

Further reading: 
https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#awaiting_a_promise.all